httpx/_client.py

```diff
@@ -543,10 +543,13 @@ def _transport_for_url(self, url: URL) -> httpcore.SyncHTTPTransport:
         enforce_http_url(url)
 
         if self._proxies and not should_not_be_proxied(url):
-            is_default_port = (url.scheme == "http" and url.port == 80) or (
-                url.scheme == "https" and url.port == 443
+            default_port = {"http": 80, "https": 443}[url.scheme]
+            is_default_port = url.port is None or url.port == default_port
+            hostname = (
+                f"{url.host}:{default_port}"
+                if url.port is None
+                else f"{url.host}:{url.port}"
             )
```

Comment:
Could this be simplified:
```suggestion
			port = url.port or default_port
            hostname = f"{url.host}:{port}"
```

Reply:
Great call yup.